### PR TITLE
windows: download appveyor release artifacts automatically

### PR DESCRIPTION
**Pull request by kenhys**
It download artifacts files and rename to distinct filename.

### CODE REVIEW

**Comment by kenhys** [comment]
@kou Thanks, I've fixed to use `veyor` gem. (it can fetch job_id correctly)

File: packages/windows/download_appveyor_build.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    suffix << "-vs2013"
+  end
+  artifacts.each do |artifact|
+    file_name = artifact["fileName"]
+    url = "#{appveyor_url}api/buildjobs/#{job_id}/artifacts/#{file_name}"
+    file_name  = artifact["fileName"].sub(/#{arch}/, suffix)
+    system("curl", "--location", "--output", "files/#{file_name}", url)
+    options = {
+      :content_type => "application/zip",
+    }
```
Comment (kou) [inline packages/windows/download_appveyor_build.rb]: Where is this used?

File: packages/windows/download_appveyor_build.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+base_version = (top_dir + "base_version").read.strip
+tag_name = "v#{base_version}"
+
+appveyor_url = "https://ci.appveyor.com/"
+appveyor_info = nil
+client.statuses(groonga_repository, tag_name).each do |status|
+  next unless status.target_url.start_with?(appveyor_url)
+  case status.state
+  when "pending"
+  # Ignore
```
Comment (kou) [inline packages/windows/download_appveyor_build.rb]: Need to indent.

File: packages/windows/download_appveyor_build.rb [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    suffix << "-vs2017"
+  when /VS_VERSION=14/
+    suffix << "-vs2015"
+  when /VS_VERSION=12/
+    suffix << "-vs2013"
+  end
+  artifacts.each do |artifact|
+    file_name = artifact["fileName"]
+    url = "#{appveyor_url}api/buildjobs/#{job_id}/artifacts/#{file_name}"
+    file_name  = artifact["fileName"].sub(/#{arch}/, suffix)
```
Comment (kou) [inline packages/windows/download_appveyor_build.rb]: `sub(/#{Regexp.escape(arch)}/, ...)` or `sub(arch, ...)`

Comment (kou) [reply packages/windows/download_appveyor_build.rb]: `local_file_name = file_name.sub(...)`

Comment (kou) [reply packages/windows/download_appveyor_build.rb]: How about using `groonga-${VERSION}-${ARCHITECTURE}-${VISUAL_STUDIO_VERSION}.zip` for file name generated by CMake?

File: packages/windows/download_appveyor_build.rb [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    system("curl", "--location", "--output", "files/#{file_name}", url)
```
Comment (kou) [inline packages/windows/download_appveyor_build.rb]: Using `open-uri` is better to reduce `curl` dependency.
We don't need workaround for MinGW because we run this only on GNU/Linux.

File: packages/windows/download_appveyor_build.rb [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+require "veyor"
+require "octokit"
+
+def env_value(name)
+  value = ENV[name]
+  raise "Specify #{name} environment variable" if value.nil?
+  value
+end
+
+client = Octokit::Client.new(:access_token => env_value("GITHUB_TOKEN"))
```
Comment (kou) [inline packages/windows/download_appveyor_build.rb]: GitHub token is needless because `groonga/groonga` is a public repository.

File: packages/windows/download_appveyor_build.rb (hunk 1 above)
Comment (kenhys) [reply packages/windows/download_appveyor_build.rb]: It should be removed.

File: packages/windows/download_appveyor_build.rb (hunk 3 above)
Comment (kenhys) [reply packages/windows/download_appveyor_build.rb]: It seems that good idea to contain VISUAL_STUDIO_VERSIOn in archive name. I'll change it.

File: packages/windows/download_appveyor_build.rb (hunk 5 above)
Comment (kenhys) [reply packages/windows/download_appveyor_build.rb]: Oh, surely no need to use GITHUB_TOKEN. I'll fix.

Comment (kenhys) [reply packages/windows/download_appveyor_build.rb]: removed.

File: packages/windows/download_appveyor_build.rb (hunk 4 above)
Comment (kenhys) [reply packages/windows/download_appveyor_build.rb]: fixed.

File: packages/windows/download_appveyor_build.rb (hunk 3 above)
Comment (kenhys) [reply packages/windows/download_appveyor_build.rb]: fixed to use "fileName" (download zip as is)

File: packages/windows/download_appveyor_build.rb (hunk 2 above)
Comment (kenhys) [reply packages/windows/download_appveyor_build.rb]: fixed.

File: packages/windows/download_appveyor_build.rb (hunk 1 above)
Comment (kenhys) [reply packages/windows/download_appveyor_build.rb]: removed.

File: packages/windows/download_appveyor_build.rb [hunk 6]
```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env ruby
+
+require "pathname"
+require "open-uri"
+require "veyor"
+require "octokit"
+
+def download(url, local_file)
+  ssl_verify_mode = nil
```
Comment (kou) [inline packages/windows/download_appveyor_build.rb]: Why is this needed?

File: packages/windows/download_appveyor_build.rb [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  open(url, :ssl_verify_mode => ssl_verify_mode) do |remote_zip|
```
Comment (kou) [inline packages/windows/download_appveyor_build.rb]: `remote_file`.

File: packages/windows/download_appveyor_build.rb [hunk 8]
```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env ruby
+
+require "pathname"
+require "open-uri"
+require "veyor"
+require "octokit"
+
+def download(url, local_file)
```
Comment (kou) [inline packages/windows/download_appveyor_build.rb]: `local_path`

File: packages/windows/download_appveyor_build.rb [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+  ssl_verify_mode = nil
+  open(url, :ssl_verify_mode => ssl_verify_mode) do |remote_zip|
+    File.open(local_file, "wb") do |local_zip|
```
Comment (kou) [inline packages/windows/download_appveyor_build.rb]: `local_zip` -> `local_file`

File: packages/windows/download_appveyor_build.rb (hunk 6 above)
Comment (kenhys) [reply packages/windows/download_appveyor_build.rb]: Thanks, there is no need to use it. removed.

File: packages/windows/download_appveyor_build.rb (hunk 7 above)
Comment (kenhys) [reply packages/windows/download_appveyor_build.rb]: fixed.

File: packages/windows/download_appveyor_build.rb (hunk 8 above)
Comment (kenhys) [reply packages/windows/download_appveyor_build.rb]: fixed.

File: packages/windows/download_appveyor_build.rb (hunk 9 above)
Comment (kenhys) [reply packages/windows/download_appveyor_build.rb]: fixed.
